### PR TITLE
bigtable: add support for instance state and labels

### PR DESCRIPTION
**Pull request by na-ga**
Update #2472 

- Add support for get instance state and labels
- Add support for create/update instance with labels

### CODE REVIEW

File: bigtable/admin.go [hunk 1]
```diff
@@ -714,6 +727,7 @@ func (iac *InstanceAdminClient) CreateInstance(ctx context.Context, conf *Instan
 		InstanceID:   conf.InstanceId,
 		DisplayName:  conf.DisplayName,
 		InstanceType: conf.InstanceType,
+		Labels:       conf.Labels,
```
[the remark this answers is not in the document]
Comment (na-ga) [reply bigtable/admin.go:730]: For backward compatibility, If conf.Labels is a nil map, [do not add labels to field mask](https://github.com/googleapis/google-cloud-go/blob/dc2e58d8dc2d5c72eb85f60718ce476c0ce3379c/bigtable/admin.go#L795-L798). If conf.Labels is an empty map, an existing label will be removed.

File: bigtable/admin.go [hunk 2]
```diff
@@ -671,6 +671,15 @@ func storageTypeFromProto(st btapb.StorageType) StorageType {
 	return SSD
 }
 
+// InstanceState is the state of the instance
+type InstanceState int32
+
+const (
```
[the remark this answers is not in the document]
Comment (na-ga) [reply bigtable/admin.go:677]: @tritone 
Thanks review! The instance state is used in the output only. The default value depends on the state of the bigtable instance.... What do you think?

- https://cloud.google.com/bigtable/docs/reference/admin/rpc/google.bigtable.admin.v2#instance

[the remark this answers is not in the document]
Comment (tritone) [reply bigtable/admin.go:677]: Ah, makes sense. If you could just note in the comment that it is only used in the output, that would be great.

File: bigtable/integration_test.go [hunk 3]
```diff
@@ -1259,6 +1260,7 @@ func TestIntegration_AdminCreateInstance(t *testing.T) {
 		DisplayName:  "test instance",
 		Zone:         instanceToCreateZone,
 		InstanceType: DEVELOPMENT,
+		Labels:       map[string]string{"test-label-key": "test-label-value"},
```
[the remark this answers is not in the document]
Comment (tritone) [reply bigtable/integration_test.go:1263]: Great!

File: bigtable/admin.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (tritone) [reply bigtable/admin.go:730]: Sounds good, thanks for confirming.

File: bigtable/integration_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (tritone) [reply bigtable/integration_test.go:1263]: @kolea2 could you take a look at the change to the test and see if it makes sense?

File: bigtable/admin.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (kolea2) [reply bigtable/admin.go:730]: Sounds good - can we also add this as an edge case to the existing test? One with an empty map, and one with a nil map?

File: bigtable/admin.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (tritone) [reply bigtable/admin.go:677]: Also, our linting requires a comment on the exported constants, which is why presubmits are failing currently:

```
bigtable/admin.go:678:2: exported const NOT_KNOWN should have comment (or a comment on this block) or be unexported
```

File: bigtable/admin.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (na-ga) [reply bigtable/admin.go:733]: That's great advice. Added a test case to update the instance labels. As the existing test case is not clear, I have added a new TestIntegration_ Added AdminUpdateInstanceLabels https://github.com/googleapis/google-cloud-go/pull/2477/commits/fa6130a3c5827122e955b30320fb5dd7cc5f8771. This is the result of the run from my local environment.

```
% go test -v -run TestIntegration_AdminUpdateInstanceLabels -it.use-prod -it.project my-project -it.instance a -it.table b -it.instance-to-create test-instance
=== RUN   TestIntegration_AdminUpdateInstanceLabels
=== RUN   TestIntegration_AdminUpdateInstanceLabels/update_labels
=== RUN   TestIntegration_AdminUpdateInstanceLabels/update_multiple_labels
=== RUN   TestIntegration_AdminUpdateInstanceLabels/not_update_existing_labels
=== RUN   TestIntegration_AdminUpdateInstanceLabels/delete_labels
--- PASS: TestIntegration_AdminUpdateInstanceLabels (6.88s)
    --- PASS: TestIntegration_AdminUpdateInstanceLabels/update_labels (0.82s)
    --- PASS: TestIntegration_AdminUpdateInstanceLabels/update_multiple_labels (1.23s)
    --- PASS: TestIntegration_AdminUpdateInstanceLabels/not_update_existing_labels (1.23s)
    --- PASS: TestIntegration_AdminUpdateInstanceLabels/delete_labels (0.59s)
PASS
ok  	cloud.google.com/go/bigtable	7.173s
```

File: bigtable/admin.go [hunk 4]
```diff
@@ -671,6 +671,18 @@ func storageTypeFromProto(st btapb.StorageType) StorageType {
 	return SSD
 }
 
+// InstanceState is the state of the instance. This is output-only.
+type InstanceState int32
+
+const (
+	// NOT_KNOWN is state of the instance could not be determined
```
Comment (codyoss) [inline bigtable/admin.go:678]: nit: add a period to all of these. https://github.com/golang/go/wiki/CodeReviewComments#comment-sentences

Comment (codyoss) [reply bigtable/admin.go:678]:
```suggestion
	// NOT_KNOWN represents the state of an instance that could not be determined.
```

File: bigtable/admin.go [hunk 5]
```diff
@@ -671,6 +671,18 @@ func storageTypeFromProto(st btapb.StorageType) StorageType {
...
+	NOT_KNOWN InstanceState = InstanceState(btapb.Instance_STATE_NOT_KNOWN)
+	// READY is state of the instance has been successfully created
```
Comment (codyoss) [inline bigtable/admin.go:680]:
```suggestion
	// READY represents the state of an instance that has been successfully created.
```

File: bigtable/admin.go [hunk 6]
```diff
@@ -671,6 +671,18 @@ func storageTypeFromProto(st btapb.StorageType) StorageType {
...
+	READY = InstanceState(btapb.Instance_READY)
+	// CREATING is state of the instance is currently being created
```
Comment (codyoss) [inline bigtable/admin.go:682]:
```suggestion
	// CREATING represents the state of an instance that is currently being created.
```

File: bigtable/integration_test.go [hunk 7]
```diff
@@ -1272,16 +1274,20 @@ func TestIntegration_AdminCreateInstance(t *testing.T) {
 
 	// Basic return values are tested elsewhere, check instance type
 	if iInfo.InstanceType != DEVELOPMENT {
-		t.Fatalf("Instance is not DEVELOPMENT: %v", err)
+		t.Fatalf("Instance is not DEVELOPMENT: %v", iInfo.InstanceType)
+	}
+	if got, want := iInfo.Labels, conf.Labels; !reflect.DeepEqual(got, want) {
```
Comment (codyoss) [inline bigtable/integration_test.go:1279]: Can we use `cmp.Equal` or `cmp.Diff` instead of `reflect. DeepEqual` throughout.

File: bigtable/admin.go [hunk 8]
```diff
@@ -671,6 +671,18 @@ func storageTypeFromProto(st btapb.StorageType) StorageType {
 	return SSD
 }
 
+// InstanceState is the state of the instance. This is output-only.
+type InstanceState int32
+
+const (
+	// NOT_KNOWN is state of the instance could not be determined
+	NOT_KNOWN InstanceState = InstanceState(btapb.Instance_STATE_NOT_KNOWN)
```
Comment (codyoss) [inline bigtable/admin.go:679]: MixedCaps: https://github.com/golang/go/wiki/CodeReviewComments#mixed-caps

```suggestion
	NotKnown InstanceState = InstanceState(btapb.Instance_STATE_NOT_KNOWN)
```

File: bigtable/admin.go [hunk 9]
```diff
@@ -671,6 +671,18 @@ func storageTypeFromProto(st btapb.StorageType) StorageType {
...
+	// READY is state of the instance has been successfully created
+	READY = InstanceState(btapb.Instance_READY)
```
Comment (codyoss) [inline bigtable/admin.go:681]:
```suggestion
	Ready = InstanceState(btapb.Instance_READY)
```

File: bigtable/admin.go [hunk 10]
```diff
@@ -671,6 +671,18 @@ func storageTypeFromProto(st btapb.StorageType) StorageType {
...
+	// CREATING is state of the instance is currently being created
+	CREATING = InstanceState(btapb.Instance_CREATING)
```
Comment (codyoss) [inline bigtable/admin.go:683]:
```suggestion
	Creating = InstanceState(btapb.Instance_CREATING)
```

**Comment by na-ga** [comment]
@codyoss Thanks for the review. I fixed it. Also, I have confirmed that the modified test passes from my local environment.
